File: Code/Chapter1/bubble_sort.c
/**********************
copyRight by heibanke
***********************/

#include <stdio.h>

void bubble_sort(int nums[],int length){
    int temp;
    for(int i=1;i<length;i++){
        for(int j=0;j<length-i;j++)
        {
            if(nums[j]>nums[j+1]){
                temp=nums[j];
                nums[j]=nums[j+1];
                nums[j+1]=temp;
            }
        }
    }
}

int main(){
    int aa[6]={12,23,1,4,16,34};
    bubble_sort(aa,6);
    for(int i=0;i<6;i++)
        printf("%d,",aa[i]);
    printf("\n");
    return 0;
}

File: Code/Chapter1/bubble_sort.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#copyRight by heibanke


#bubbleSort algorithm
def bubbleSort(nums):
    for j in range(len(nums),-1,-1):
        for i in range(0,j-1,1):
            if nums[i] > nums[i+1]:
                nums[i],nums[i+1] = nums[i+1],nums[i]


#test
if __name__=='__main__':
	numbers=[[9,23,12,32,12],['2', '3', '3', '6'],['b','w','u']]
	for num in numbers:
	    bubbleSort(num)
	    print(num)
